src/gpu/intel/jit/ir/message.hpp

```diff
@@ -117,9 +118,9 @@ struct block_2d_info_t {
     }
 
     // Encoded in header.
-    int surface_width = 0;
-    int surface_height = 0;
-    int surface_pitch = 0;
+    expr_t surface_width;
+    expr_t surface_height;
+    expr_t surface_pitch;
```

Comment:
Is the idea here to allow these surface parameters to be created as part of the JIT compilation (i.e. stored in registers and mutated at runtime)? If so, in what cases do we need this?

Reply:
This is needed for reusable kernels. Say we are targeting the GEMM A matrix with dimension `m` and `k`,  layout `ab` and  stride `lda`. When performing 2D loads from this matrix, we would set `surface_width = k`, `surface_height = m`, and `surface_pitch = lda` in the message header. Previously, this interface was not used for reusable kernels, hence the hard-coded surface values.